python/pyarrow/error.pyx

```diff
@@ -19,13 +19,52 @@ from pyarrow.includes.libarrow cimport CStatus
 from pyarrow.includes.common cimport c_string
 from pyarrow.compat import frombytes
 
+
 class ArrowException(Exception):
     pass
 
+
+class ArrowInvalid(ValueError, ArrowException):
```

Comment:
We do have a couple of these in pandas: https://github.com/pandas-dev/pandas/blob/master/pandas/errors/__init__.py